### DISCUSSION TITLE
IRAF/X11 install issues on WSL2/Ubuntu

### DISCUSSION THREAD

**Discussion by cphenr02** [iraf-community/iraf#182] [Q&A]
Hello,

I have been attempting to install IRAF and X11IRAF to my Windows 10 device with Ubuntu 20.04 (using WSL2) but keep running into issues. The primary issue is while attempting to compile X11 and occurs when attempting to compile ximtool specifically.

I installed IRAF using `sudo apt install iraf` which appeared to work fine however I receive an error saying 

> Command 'xgterm' not found, did you mean:
command 'xterm' from deb xterm (353-1ubuntu.1.20.04.2)
Try: sudo apt install <deb name>

I believe these is due to needed X11 however this runs into the issue mentioned above.

I've attached a log of the compilation process including errors. I'm very new to Linux but need IRAF for my internship so if anyone could walk me through how to resolve these issues I would greatly appreciate it. If a build.log is needed, can you please direct me on how to pull this as well? (again, new to Linux! Sorry!)

[x11_make_error.txt.txt](https://github.com/iraf-community/iraf/files/6696871/x11_make_error.txt.txt)

Thank you in advance.

**Comment by olebole** [comment]
If you just need xgterm, then it is already compiled now as `xgterm/xgterm`. Copy it to `/usr/local/bin/`, and you are ready for IRAF.

The error you see happened when you tried to compile an optional IRAF extension for ximtool. So, ximtool itself should also already usable as `ximtool/ximtool` (without the extension doesn't display the world coordinates).

This error looks as if you had one of the IRAF environment variables set from an earlier attempt to compile/install IRAF, like `iraf`, `host` or so. Please check also that the file `/usr/lib/iraf/extern/.zzsetenv.def` exists (maybe an empty file), and that you used the correct `xc` (the one that comes with Ubuntu, in `/usr/bin`).

However, I strongly recommend to not start with IRAF. IRAF is an old, abandoned piece of software, which is kept here mainly for people who still rely on this.
Instead I recommend to use [astropy](https://astropy.org) and its affiliated packages.
